### ISSUE TITLE
请教一下tsne

### ISSUE THREAD

**Issue by ChongLi2001** [XLearning-SCU/2024-ICML-TAC#9]
[大佬，请教一下您的tsne图。我画出来的都是细长形状的，但看到您的论文里有"Visualization of features extracted by different methods on the ImageNet-Dogs training set, with the corresponding k-means clustering ARI annotated on the top."这句话，想问问您有画tsne的代码吗？是需要提取feature后进行kmeans，再画tsne图吗？

![Image](https://github.com/user-attachments/assets/dc1149b7-235f-4165-a0a0-84e19f114bee)

**Comment by Yunfan-Li** [comment]
您好，不需要用kmeans，直接对feature做tsne就行了，我用的是sklearn的tsne（https://scikit-learn.org/stable/modules/generated/sklearn.manifold.TSNE.html ）

**Comment by ChongLi2001** [comment]
#加载
 model=torch.load("xxx")

dataset_image_train = TensorDataset(torch.from_numpy(images_embedding_train).float())

dataloader_train = DataLoader(dataset_image_train, batch_size=512, shuffle=False)

preds, features= infer(model, dataloader_train)

plot_tsne(features, preds)

那请问是这个逻辑吗
